core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -575,27 +575,35 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
   private def changeToWitnessUTXO(
       transactionOutput: TransactionOutput): InputPSBTMap = {
     val newElements = transactionOutput.scriptPubKey match {
-      case _: WitnessScriptPubKey =>
+      case rs: WitnessScriptPubKey
+          if !rs.isInstanceOf[BIP143VulnerableScriptPubKey] =>
         filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
       case _: P2SHScriptPubKey =>
-        if (redeemScriptOpt.isDefined && redeemScriptOpt.get.redeemScript
-              .isInstanceOf[WitnessScriptPubKey]) {
-          filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
+        if (redeemScriptOpt.isDefined) {
+          val redeemScript = redeemScriptOpt.get.redeemScript
+          if (redeemScript
+                .isInstanceOf[WitnessScriptPubKey] && !redeemScript
+                .isInstanceOf[BIP143VulnerableScriptPubKey]) {
+            filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
```

Comment:
this case isn't possible, no need to complicate things

Reply:
I think this is fine to keep for now, it should never be ran currently because there is only `WitnessScriptPubKeyV0`s, and come taproot we'll just have to add this back in